include/drivers/dma.h

```diff
@@ -327,6 +341,52 @@ static inline int z_impl_dma_stop(const struct device *dev, uint32_t channel)
 	return api->stop(dev, channel);
 }
 
+/**
+ * @brief request DMA channel.
+ *
+ * request DMA channel resources
+ * return -EINVAL if there is no valid channel available.
+ *
+ * @param dev     Pointer to the device structure for the driver instance.
+ * @param filter  channel filter
+ *
+ * @retval dma channel if successful.
+ * @retval Negative errno code if failure.
+ */
+__syscall int dma_request_channel(const struct device *dev,
+				  enum dma_channel_filter filter);
+
+static inline int z_impl_dma_request_channel(const struct device *dev,
```

Comment:
Should these APIs be considered optional?

Should this be:
```
if (api->request_channel) {
   return api->request_channel(dev, filter);
}

return -ENOTSUP;
```

Reply:
good idea, updated